src/explore-education-statistics-admin/src/pages/release/edit-release/content/helpers.ts

```diff
@@ -0,0 +1,345 @@
+import { KeyStatsFormValues } from '@admin/modules/editable-components/EditableKeyStatTile';
+import permissionsService from '@admin/services/permissions/service';
+import {
+  EditableContentBlock,
+  ExtendedComment,
+} from '@admin/services/publicationService';
+import { releaseContentService } from '@admin/services/release/edit-release/content/service';
+import {
+  ContentBlockAttachRequest,
+  ContentBlockPostModel,
+} from '@admin/services/release/edit-release/content/types';
+import { Dictionary } from '@admin/types';
+import {
+  AbstractRelease,
+  ContentSection,
+} from '@common/services/publicationService';
+import { AxiosResponse } from 'axios';
+import { RemoveBlockFromSection } from './actions';
+import { Dispatch } from './ReleaseContext';
+
+const contentSectionComments = (
+  contentSection?: ContentSection<EditableContentBlock>,
+) => {
+  if (
+    contentSection &&
+    contentSection.content &&
+    contentSection.content.length > 0
+  ) {
+    return contentSection.content.reduce<ExtendedComment[]>(
+      (allCommentsForSection, content) =>
+        content.comments
+          ? [...allCommentsForSection, ...content.comments]
+          : allCommentsForSection,
+      [],
+    );
+  }
+
+  return [];
+};
+
+const getUnresolveComments = (release: AbstractRelease<EditableContentBlock>) =>
+  [
+    ...contentSectionComments(release.summarySection),
+    ...contentSectionComments(release.keyStatisticsSection),
+    ...release.content
+      .filter(_ => _.content !== undefined)
+      .reduce<ExtendedComment[]>(
+        (allComments, contentSection) => [
+          ...allComments,
+          ...contentSectionComments(contentSection),
+        ],
+        [],
+      ),
+  ].filter(comment => comment !== undefined && comment.state === 'open');
+
```

Comment:
The functions below here are all calling out to be composed into a hook called `useReleaseActions`.

```typescript
export default function useReleaseActions() {
  const dispatch = useReleaseDispatch();

  const getReleaseContent = useCallback(async (releaseId: string) => {
    // ...
  }, [dispatch]);

  return {
    getReleaseContent,
    // ...the other functions
  }
}
```

This way we don't need to pass in a `dispatch` every single time.

We don't need to be adding in a param for the error handler as this should just be chained on at the call sites either with `try/catch` or `.catch` (if using a promise).

Reply:
Really nice! just put this in! :)